symtab.h

```diff
@@ -66,6 +66,10 @@ inline bool operator==(const Symbol& x, const Symbol& y) {
   return x.val() == y.val();
 }
 
+inline bool operator<(const Symbol& x, const Symbol& y) {
+  return x.str() < y.str();
```

Comment:
As long as val is a consistent order run to run, that's fine too. I just made this change, and didn't notice any large slowdowns when parsing the android tree. I didn't actually do any benchmarking.